sklego/linear.py

```diff
@@ -0,0 +1,140 @@
+
+    def __init__(self, weighting_method: Union[str, None] = None, span: float = .1):
+        super().__init__()
+
+        self._check_init_inputs(weighting_method, span)
+
+        self.weighting_method = weighting_method
+        self.span = span
+        self.xs = None
+        self.ys = None
+        self.dim_ = None
+        self.logger = logging.getLogger(__name__)
+
+    def fit(self, X: np.array, y: np.array):
+        """
+        Fit the regressor on the data by storing the data sorted on the inputs.
+
+        :param X: Array-like object of shape (n_samples, 1), input data for the model.
+        :param y: Array-like object of shape (n_samples, 1), target output data for the model
+        :return: The fitted instance
+        """
+
+        self.xs, self.ys = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+        self.dim_ = self.xs.shape[1]
+
+        return self
+
+    def predict(self, X: np.array, with_indices: bool = False) -> Union[np.array,
+                                                                        (np.array, np.array)]:
+        """
+        Predict targets using the fitted model
+        :param X: Array-like object of shape (n_samples, 1), input data for the model.
+        :param with_indices: If set to True, function also returns the indices of the sub-set in the
+            train data used for fitting the local model.
+        :return: Array-like object of shape (n_samples, 1), predicted target output. If with_indices
+            is set to true, a tuple of two array-like objects (y_predictions, indices) is returned
+            instead.
+        """
+
+        check_is_fitted(self, ['dim_'])
+        X = check_array(X)
+
+        y_pred = np.array([])
+
+        if with_indices:
+            indices = []
+
+        for x in X:
+            idx_window = self._get_window_indices(x.reshape(-1, 1))
+
+            if with_indices:
+                indices.append(idx_window)
+
+            X = self.xs[idx_window].reshape(-1, 1)
+            y = self.ys[idx_window]
+
+            weights = self._create_weights(x, X)
+            model = LinearRegression().fit(X, y, sample_weight=weights)
+
+            y_pred = np.append(y_pred, model.predict(x.reshape(-1, 1)))
+
+        if with_indices:
+            return y_pred, indices
+        else:
+            return y_pred
+
+    @staticmethod
+    def _check_init_inputs(weighting_method: str, span: float) -> None:
+        """
+        Checks if the provided model parameters are valid.
+
+        :param weighting_method: String id of the weighting method to be used. Should be 'euclidean'
+            or 'equal'.
+        :param span: Float in (0,1]. Sets the fraction of data to use for making local predictions.
+            If 1, the full training data set is used.
+        """
+
+        if not 0 < span <= 1:
+            raise ValueError("Span should be larger than 0 and smaller or equal to 1")
+
+        expected_weighting_methods = ['euclidean', None]
```

Comment:
why not add this to `self`?

Reply:
also, why is `None` a setting?

Reply:
Changed 'None' to 'unweighted' to make it more clear that will make the algorithm not use any weights.